New blog post recapping the April sprint (#407)
temporalio/documentation: blog/2021-05-06-documentation-information-architecture-sprint.md

---
tags:
  - documentation
  - information-architecture
  - temporal
  - site design
posted_on_: 2021-05-06T00:00:00Z
slug: docs-info-arch-2021

> title: 'Docs site April 2021 sprint'

Reviewer: what are your thoughts on titling something more generally useful to others? imagine someone copy pasting in to their Slack, saying "we need to do this" - what does the title of that blogpost say?

```suggestion
title: 'Write Docs for Clear User Personas'
```

However, one must learn not only of Temporal's overall strategy, but also language specific implementation details simultaneously to get a taste of that secret sauce.
At the time of writing this, we supported three implementation languages with a fourth on the way, with each language presenting its own unique way of defining and writing Workflows.
My instinct was that something fundamental was off in the current approach to presenting this information.

I needed to think.
So, I went for a run.

![Run gif](https://media.giphy.com/media/13ZPuK9cmWembe/giphy.gif)

With my legs pumping freshly oxygenated blood through my brain, my heart, and my veins, I set my mind to the task at hand.
I tried to come at the problem from a new user's perspective.
This is how I discovered a problem that I hadn't fully noticed before.

## User personas

The thing is, not all users are after the same information.
Unlike many other technologies, we don't offer a single service or API endpoint.
Temporal's boundaries are high and wide, and there are at least four unique user groups accessing the documentation.

> 1. Infrastructure and operations engineers looking to deploy and operate their own Temporal Server cluster.
> 2. Developers and software engineers looking to start building applications.
> 3. Operators and SREs looking to troubleshoot, debug, or generally monitor the system's health.
> 4. Business decision makers evaluating the technology as a whole or considering the use of our cloud service (still in beta).

Reviewer: ```suggestion
1. **Infrastructure and operations engineers** looking to deploy and operate their own Temporal Server cluster.
2. **Developers and software engineers** looking to start building applications.
3. **Operators and SREs** looking to troubleshoot, debug, or generally monitor the system's health.
4. **Business decision makers** evaluating the technology as a whole or considering the use of our cloud service (still in beta).
```

Thus, our core concepts section would become a sort directory for each of the languages, directing developers to language specific concept pages.

## April 2021 changes

In April, the Temporal Product Team set about iterating towards these solutions.
And we are excited with the results!

We redesigned our documentation landing page, [docs.temporal.io](/):

![Temporal documentation landing page April 2021](/img/documentation-landing-page-april-2021.png)

We implemented a new SDK landing page, [docs.temporal.io/application-development](/application-development):

![Temporal documentation app dev landing page April 2021](/img/documenation-app-dev-landing-page-april-2021.png)

And we created a much more immersive experience for the user personas that we identified above:

- Go developers for example, [docs.temporal.io/go](/docs/go/introduction):

![Go topic area](/img/go-topic-area.png)

- Infra engineers for example, [docs.temporal.io/server](/docs/server/introduction):

![Server topic area](/img/server-topic-area.png)

And we started converting the core concepts into a directory for new users learning about Temporal:

![Core concepts area](/img/concepts-topic-area.png)

> There is a lot more work to be done in each of the language areas to ensure full coverage of the concepts within each language.
> But now we feel there is an information architecture in place that fundamentally addresses our user's personas such learning about how and why of a particular concept becomes a much more rewarding and informative experience.

Reviewer: more active voice and concluding section heading

```suggestion

## Still Day One

We have finally arrived at a place where there is an information architecture that fundamentally matches our user's personas. 
With this user focus, our hope is that learning about the how and why of a particular concept becomes a much more rewarding and informative experience.

We know we have a LOT more work to do. We still lack full coverage of the concepts within each language, and the navigation and search experience can be made more intuitive. If you have ideas for how to improve them, [email us](mailto:docs@temporal.io)!
```
